### PR TITLE
Return errors using RunE

### PR DESCRIPTION
**Pull request by RaghavRoy145**
This PR is related to this issue: https://github.com/dims/maintainers/issues/12

**Changes include:**
- Using RunE instead of Run for all commands
- Returning errors instead of panic in the RunE implementation

### CODE REVIEW

**Comment by RaghavRoy145** [comment]
For this change, I have removed the error message that was part of the panic and instead just return the error directly, If the error needs to be returned like the previous implementation (strings describing the error using Errorf ) I can push that commit as well.

File: cmd/audit.go [hunk 1]
```diff
@@ -74,18 +75,19 @@ var auditCmd = &cobra.Command{
 
 		sigsYamlPath, err := utils.GetSigsYamlFile(pwd)
 		if err != nil {
-			panic(fmt.Errorf("ERROR: unable to find sigs.yaml file: %w", err))
+			return err
 		}
 		context, err := utils.GetSigsYaml(sigsYamlPath)
 		if err != nil {
-			panic(fmt.Errorf("ERROR: parsing file: %s - %w", sigsYamlPath, err))
+			return err
```
Comment (MadhavJivrajani) [inline cmd/audit.go:82]: No strong opinion on retaining the format string, will defer to @\nikhita, @\palnabarun and @\dims

File: cmd/validate.go [hunk 2]
```diff
@@ -79,6 +79,7 @@ var validateCmd = &cobra.Command{
 			}
 			//panic("please see errors above")
 		}
+		return err
```
Comment (MadhavJivrajani) [inline cmd/validate.go]: Which `err` is this supposed to be?

**Review by MadhavJivrajani — COMMENTED** [review]
Left a comment.
Also, can you please squash commits?
Thanks!

File: cmd/audit.go (hunk 1 above)
Comment (RaghavRoy145) [reply cmd/audit.go:82]: This function expected a return statement at the end, so I've returned `err` that was last assigned, is there a better way to take care of that?

File: cmd/validate.go (hunk 2 above)
Comment (RaghavRoy145) [reply cmd/validate.go]: This function expected a return statement at the end, so I've returned `err` that was last assigned, is there a better way to take care of that, like returning `nil` ?

Comment (MadhavJivrajani) [reply cmd/validate.go]: Yeah now that I think about it, returning `nil` here would make more sense considering we are just printing out warnings in the current implementation and not actually `panic`ing.

**Review by MadhavJivrajani — APPROVED** [review]
Squash again if possible! :)
LGTM from me.

Something to consider tho:
If a command errors, `cobra` typically prints out the error twice in the following manner:
```
❯ maintainers check-urls
Running script : 02-17-2022 18:25:02
Processing sigs.yaml
Error: open sigs.yaml: no such file or directory
Usage:
  maintainers check-urls [flags]

Flags:
  -h, --help               help for check-urls
      --yaml-file string   validate urls in this yaml file (default "sigs.yaml")

Error: open sigs.yaml: no such file or directory
```
This is true for all commands, this is 100% a nit, but it nagged me a little when I saw that :p 

Anyways, if we want to fix it, we can do so by setting `SilenceErrors` to `true` for commands.
For ex, for the `check-urls` commands:
```go
func init() {
	checkURLsCmd.Flags().StringVar(&yamlFile, "yaml-file", "sigs.yaml", "validate urls in this yaml file")
	checkURLsCmd.SilenceErrors = true
	rootCmd.AddCommand(checkURLsCmd)
}
```
and what you'll get is:
```
Running script : 02-17-2022 18:29:29
Processing sigs.yaml
Usage:
  maintainers check-urls [flags]

Flags:
  -h, --help               help for check-urls
      --yaml-file string   validate urls in this yaml file (default "sigs.yaml")

Error: open sigs.yaml: no such file or directory
```
(if we do this, it can be a separate PR :))

**Pull request merged**